src/core/predicate/base.rs

```diff
@@ -36,11 +40,11 @@ impl<P, PI, H, HI> Handler<(PI, HI)> for Predicate<P, PI, H, HI>
 where
     P: Handler<PI> + Clone + 'static,
     P::Output: Into<PredicateResult>,
-    PI: TryFromInput + 'static,
+    PI: TryFromInput + Clone + 'static,
```

Comment:
Lower requirements by manual Clone implementation for Predicate

Reply:
Predicate already implements Clone. Whatr do you mean?

Reply:
See my comment for `src/dialogue/mod.rs`